texastribune/queso-ui
# Docs redesign and restructure + switch to modern normalize

#### What's this PR do?

- Deletes legacy files. The time has come to move on from those (this will be a major release for that reason)
- Switches up our homespun docs approach to [11ty](https://www.11ty.dev/)


##### Classes added (if any)
- Our reset file, normalize was swapped with modern-normalize
##### Classes removed (if any)
- A BUNCH of now unused legacy files


#### Why are we doing this? How does it help us?
- Modern normalize is a little less bulky and actively maintained to eventually become obsolete as browsers improve
- More user-friendly docs (🤞 hopefully!)
- The switch to 11ty just makes it easier to spin up docs pages as we need them. This helps reduce writing our own build code for docs to rely on a better maintained system
- The legacy code drop helps eliminate some noise, reducing the need to dig through old file only to find they aren't used

#### How should this be manually tested?
`npm install`
`npm run dev`

Or just go here where this is temporarily preview-able: https://test-queso-docs.surge.sh/

**There's so much code to sift through here and unless you want, feel free to give little consideration to the docs changes. They don't affect the experience of where queso is used in our repos.**

The feedback that would be super helpful for this would be assessing the usability of the docs.
- Are the docs structured in a way that make sense to you?
- Say you wanted to see how many times the story block component is used and where. Do you know where to go and where to look for that?
- Would you know that all the github icons take you to the github instance of the CSS if I hadn't have just said so?



#### Does this introduce a breaking change where queso-ui is used in the wild? If so, is there a relevant branch/PR to accompany this release?

I deleted base-v2.scss to favor being more selective about what parts of the system you import. Also the switch to modern normalize could have some weird effects in obscure places. This will be a major release for those reasons.


#### TODOs / next steps:

* [ ] *Tear down surge site*
